### PR TITLE
Add exception binding tests

### PR DESCRIPTION
**Pull request by jslee02**

### CODE REVIEW

File: test/examples/30_pybind11_examples/02_advanced/03_exceptions/exceptions.h [hunk 1]
```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <exception>
+#include <stdexcept>
+
+#include <pybind11/pybind11.h>
+
+////////////////////////////////////////////////////////////////////////////////
+//
+// Following examples are from
```
Comment (psigen) [inline test/examples/30_pybind11_examples/02_advanced/03_exceptions/exceptions.h:10]:
```suggestion
// The following examples are from
```

File: test/examples/30_pybind11_examples/02_advanced/03_exceptions/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import unittest
+
+import exceptions_pybind11 as py11
+import exceptions_boost_python as boost
```
Comment (psigen) [inline test/examples/30_pybind11_examples/02_advanced/03_exceptions/exceptions.py:4]: What does boost_python do here?

File: test/examples/30_pybind11_examples/02_advanced/03_exceptions/exceptions.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    def _test_built_in_exception_translation(self, binding):
+        if binding is py11:
+            self.assertRaises(RuntimeError, binding.throw_std_exception)
+            self.assertRaises(MemoryError, binding.throw_bad_alloc)
+            self.assertRaises(ValueError, binding.throw_domain_error)
+            self.assertRaises(ValueError, binding.throw_invalid_argument)
+            self.assertRaises(ValueError, binding.throw_length_error)
+            self.assertRaises(IndexError, binding.throw_out_of_range)
+            self.assertRaises(ValueError, binding.throw_range_error)
+            self.assertRaises(OverflowError, binding.throw_overflow_error)
```
Comment (psigen) [inline test/examples/30_pybind11_examples/02_advanced/03_exceptions/exceptions.py:18]:
```suggestion
            self.assertRaises(OverflowError, binding.throw_overflow_error)
        # TODO: Fill in Boost.Python handling of exceptions.
```

File: test/examples/30_pybind11_examples/02_advanced/03_exceptions/exceptions.py (hunk 2 above)
Comment (jslee02) [reply test/examples/30_pybind11_examples/02_advanced/03_exceptions/exceptions.py:4]: We test both pybind11 and boost_python. See [lines 21-22](https://github.com/personalrobotics/chimera/pull/290/files#diff-7f929e495012676319cf582f7f0987f6R21-R22) in this file.

**Pull request merged**